### PR TITLE
add more init container testing

### PR DESCRIPTION
**Pull request by tzneal**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Experiment in testing container lifetimes. The intent is to try a different method that could allow for easier expansion of the testing to support sidecar containers.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/init_containers.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	lineNo := 0
+	for sc.Scan() {
+		lineNo++
+		fields := strings.Fields(sc.Text())
+		if len(fields) < 3 {
+			framework.ExpectNoError(fmt.Errorf("%v should have at least length 3", fields))
+		}
+		res = append(res, containerOutput{
+			line:          lineNo,
+			timestamp:     fields[0],
```
Comment (tzneal) [inline test/e2e_node/init_containers.go:218]: Might parse this to an int and use this at some point, busybox doesn't support %N so we only have 1 second resolution here.

Comment (xmcqueen) [reply test/e2e_node/init_containers.go:277]: This does work out to produce something that's nicely testable.

**Comment by tzneal** [comment]
> @tzneal is this ready for review? If so, I'd suggest to adjust the commit message a bit.
> 
> /kind cleanup

Yes, feedback from sidecar WG was positive. Just updated the message, I forgot how bad the original experiment message was 😄

**Comment by tzneal** [comment]
The tests ran locally when I used FOCUS, but I don't see them in the output of `pull-kubernetes-e2e-gce`.  I renamed the source file to `_test`, in case that was it, but still nothing.  What am I missing?

**Comment by marquiz** [comment]
> `pull-kubernetes-e2e-gce`. I renamed the source file to `_test`, in case that was it, but still nothing. What am I missing?

They are not run as part of that job, but look at `pull-kubernetes-node-e2e-containerd`

**Review by SergeyKanzhelev — APPROVED** [review]
I see both under here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/115984/pull-kubernetes-node-e2e-containerd/1633568222856876032/ `E2eNode Suite: [It] [sig-node] InitContainers [NodeConformance] should not launch regular containers if an init container fails`

/lgtm
/approve

**Pull request merged**